packages/core/src/routes/applications/application-protected-app-metadata.ts

```diff
@@ -51,7 +56,13 @@ export default function applicationProtectedAppMetadataRoutes<T extends AuthedRo
         'domain.limit_to_one_domain'
       );
 
-      // TODO: LOG-8066 check if domain is already in use
+      assertThat(
+        !(await findApplicationByProtectedAppCustomDomain(domain)),
+        new RequestError({
+          code: 'domain.hostname_already_exists',
+          status: 409,
```

Comment:
need to double-check,  we might use 422 on other resource exists exceptions.

Reply:
Updated with 422 to align with other exceptions.